### PR TITLE
Upstream firewalls DSF update to allow unknown values

### PR DESCRIPTION
**Pull request by slevenick**
Upstreams: https://github.com/hashicorp/terraform-provider-google/pull/10668

It's unclear to me if this is the best that we can do, it's not *quite* correct, but probably a better failure mode than exists right now

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
compute: fixed a bug where `google_compute_firewall` would incorrectly find `source_ranges` to be empty during validation
```

### CODE REVIEW

**Comment by jackwhelpton** [comment]
Based on the identified problem with this (`source_ranges` being removed), could we perhaps improve the check by using `diff.HasChange`?

**Comment by jackwhelpton** [comment]
Any progress on this? I imagine there's not much activity until after the holiday period now, but this is still a blocker for https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1071.

File: mmv1/third_party/terraform/tests/resource_compute_firewall_test.go.erb [hunk 1]
```diff
@@ -445,3 +468,40 @@ resource "google_compute_firewall" "foobar" {
...
+
+resource "google_compute_subnetwork" "foobar" {
+  name          = "%s-subnet"
+  ip_cidr_range = "10.0.0.0/16"
+  region        = "us-central1"
+  network       = google_compute_network.foobar.name
+}
+
+resource "google_compute_address" "foobar" {
+	name         = "%s-address"
```
Comment (megan07) [inline mmv1/third_party/terraform/tests/resource_compute_firewall_test.go.erb:487]: nit: the spacing in this resource is tabbed rather than 2 spaces

**Review by megan07 — APPROVED** [review]
LGTM

**Pull request merged**